connect/transforms/src/main/java/org/apache/kafka/connect/transforms/SetSchemaMetadata.java

```diff
@@ -71,7 +78,7 @@ public R apply(R record) {
                 schema.type(),
                 schema.isOptional(),
                 schema.defaultValue(),
-                schemaName != null ? schemaName : schema.name(),
+                buildSchemaName(schemaNamespace, schemaName != null ? schemaName : schema.name()),
```

Comment:
IIUC, this new `buildSchemaName` method matches the same behavior when a `schema.name` configuration value is provided, but no `schema.namespace` config value is provided: the name is used no matter what the `schema.name` value is. For example, a `schema.name=` config value would use the blank (or empty) name for the schema.

However, if such a user were to simply _add_ a `schema.namespace` config property and keep the `schema.name` set to a blank or empty string, then this method throws a `DataException`.

It might be better to try to keep the behavior as close as possible to the older behavior, except simply use a `schema.namespace` when set to a non-blank or non-empty string. IOW:
```
protected static String buildSchemaName(String namespace, String name) {
    if (Utils.isBlank(namespace)) {
        return name;
    }
    namespace = namespace.trim().replaceAll("[.]$", "");
    return namespace + "." + name;
}
```

Reply:
I agree to @pjmagee 's opinion. 

if I want to replace just namespace of multiple schemas, this is not suitable to configure.
this transform is not consider to this situation?

Reply:
@daehokimm, thank for your review, I guess [this parameterized unit-test](https://github.com/apache/kafka/pull/11442/files#diff-8470d18c341b6c20f7a6751d2f42af43560525bfeca29bd980ec29d256153205R77) should answer your question.
The case with just namespace set and an empty schema Name is acceptable.